src/validation.cpp

```diff
@@ -126,12 +130,7 @@ CScript COINBASE_FLAGS;
 
 // Internal stuff
 namespace {
-    CBlockIndex *&pindexBestInvalid = ::ChainstateActive().pindexBestInvalid;
-
-    /** All pairs A->B, where A (or one of its ancestors) misses transactions, but B has transactions.
-     * Pruned nodes may have entries where B is missing data.
-     */
-    std::multimap<CBlockIndex*, CBlockIndex*>& mapBlocksUnlinked = ::ChainstateActive().mapBlocksUnlinked;
+    CBlockIndex* pindexBestInvalid = nullptr;
```

Comment:
Why is this changed?

Reply:
Ah sorry, this was a little confusing. Because `pindexBestInvalid` is only used in validation.cpp, I removed it as a member of `CChainState` (though in this revision I forgot to actually remove the declaration - willfix) since it doesn't need to be a part of that interface. There is a change in name only, because `::ChainstateActive().pindexBestInvalid` is nullptr at the time of this declaration.

Reply:
Also worth pointing out that we need to touch this piece of data because it is used by both `CChainState` methods *and* `BlockManager` methods.

Reply:
> pindexBestInvalid just seems like it should be a BlockManager member. Or am I missing something?

In theory we could, the only problem here is that (i) [`g_blockman` is retired in later commits](https://github.com/bitcoin/bitcoin/pull/15606/commits/df81dc462e72047ed6857d927946cb18e0e9492b#diff-24efdb00bfbe56b140fb006b562cc70bL81) and (ii) pindexBestInvalid is used in non-method functions like `CheckForkWarningConditions`, so essentially there'd be no way to reference `m_blockman.pindexBestInvalid` in functions like that without moving them onto CChainState. So instead of creating extra diff, I just figured keeping pindexBestInvalid private to validation.cpp made sense. Let me know if you think it'd be better as a member of BlockManager.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/16194#discussion_r296840549

Thanks for explaining. It would be nice to clean up and document these free floating global variables (pindexBestInvalid, pindexBestForkTip, pindexBestForkBase), giving them `g_` prefixes or moving them to appropriate classes so they don't look like local variables. It's beyond the scope of this PR, though.

FWIW, I don't see later commits interfering here. It looks like there still is a single blockman instance, just called g_chainman.m_blockman instead of g_blockman.